editor/src/components/editor/actions/actions.tsx

```diff
@@ -4771,15 +4774,39 @@ export const UPDATE_FNS = {
           }
         },
       )
-      const updatedEditorState: EditorModel = {
-        ...withNewElement,
-        selectedViews: newSelectedViews,
-        trueUpGroupsForElementAfterDomWalkerRuns: [
-          ...editor.trueUpGroupsForElementAfterDomWalkerRuns,
-          EP.appendToPath(action.insertionPath.intendedParentPath, newUID),
-        ],
+
+      let groupCommands: CanvasCommand[] = []
+      if (treatElementAsGroupLike(editor.jsxMetadata, action.insertionPath.intendedParentPath)) {
+        const groupFrame = MetadataUtils.getFrameOrZeroRectInCanvasCoords(
+          action.insertionPath.intendedParentPath,
+          withNewElement.jsxMetadata,
+        )
+
+        if (action.toInsert.element.type === 'JSX_ELEMENT') {
+          // this condition would need updating when we can insert fragments or conditionals with children
+          groupCommands.push(
+            ...createPinChangeCommandsForElementInsertedIntoGroup(
+              newPath,
+              right(action.toInsert.element.props),
+              groupFrame,
+              localRectangle(groupFrame),
```

Comment:
this looks wrong! a localRectangle means local relative to the offset-providing parent or ancestor. there's functions to convert a globalFrame to a local rectangle by providing the parent. or you could just access groupMetadata.localRectangle

Reply:
aha! would this work? https://github.com/concrete-utopia/utopia/pull/4004/commits/c57499f86b8e3b69eb855b2fc3eb6de7bca88b4a